x/hard/keeper/withdraw.go

```diff
@@ -9,20 +9,25 @@ import (
 
 // Withdraw returns some or all of a deposit back to original depositor
 func (k Keeper) Withdraw(ctx sdk.Context, depositor sdk.AccAddress, coins sdk.Coins) error {
-	deposit, found := k.GetDeposit(ctx, depositor)
+	// Call incentive hooks
+	existingDeposit, found := k.GetDeposit(ctx, depositor)
 	if !found {
 		return sdkerrors.Wrapf(types.ErrDepositNotFound, "no deposit found for %s", depositor)
 	}
-	// Call incentive hooks
-	k.BeforeDepositModified(ctx, deposit)
+	k.BeforeDepositModified(ctx, existingDeposit)
+
 	existingBorrow, hasExistingBorrow := k.GetBorrow(ctx, depositor)
 	if hasExistingBorrow {
 		k.BeforeBorrowModified(ctx, existingBorrow)
 	}
 
+	// Sync interest
 	k.SyncBorrowInterest(ctx, depositor)
 	k.SyncSupplyInterest(ctx, depositor)
 
+	// Refresh Deposit after syncing interest
+	deposit, found := k.GetDeposit(ctx, depositor)
```

Comment:
```suggestion
	deposit, _ := k.GetDeposit(ctx, depositor)
```

Can ignore found since we aren't checking it (already checked)

Reply:
https://github.com/Kava-Labs/kava/pull/824/commits/8e49ed54fb8d863acf63b2e4ebb620e4105c921a